docs/doc/14-sql-commands/00-ddl/20-table/60-optimize-table.md

```diff
@@ -87,70 +95,60 @@ OPTIMIZE TABLE [database.]table_name [ PURGE | COMPACT | ALL | [SEGMENT] [LIMIT
-| f150212c302244fa9780cc6337138c6e |            16 |          16 |  10000000 |
-+----------------------------------+---------------+-------------+-----------+
-mysql> -- newly generated blocks are too small, let's compact them
-mysql> optimize table t compact;
-mysql> select snapshot_id, segment_count, block_count, row_count from fuse_snapshot('default', 't');
-+----------------------------------+---------------+-------------+-----------+
-| snapshot_id                      | segment_count | block_count | row_count |
-+----------------------------------+---------------+-------------+-----------+
-| 4f33a63031424ed095b8c2f9e8b15ecb |            16 |          16 |  10000005 | // <- the new snapshot
-| 51a03a3d0e1241529097f7a2f20b76eb |            21 |          21 |  10000005 |
-| f27c19e1c85046f6a73403de846f9483 |            20 |          20 |  10000004 |
-| 9baba3bbe0244aab9d1b25f5b69f7378 |            19 |          19 |  10000003 |
-| d90010f11b574b078becbc6b6791703e |            18 |          18 |  10000002 |
-| f1c1f24d9d3742b9a05c827a3f03e0e0 |            17 |          17 |  10000001 |
-| f150212c302244fa9780cc6337138c6e |            16 |          16 |  10000000 |
-+----------------------------------+---------------+-------------+-----------+
-
-mysql> -- If we do not care about the historical data
-mysql> optimize table t purge;
-Query OK, 0 rows affected (0.08 sec)
-Read 0 rows, 0.00 B in 0.041 sec., 0 rows/sec., 0.00 B/sec.
-
-mysql> select snapshot_id, segment_count, block_count, row_count from fuse_snapshot('default', 't');
-+----------------------------------+---------------+-------------+-----------+
-| snapshot_id                      | segment_count | block_count | row_count |
-+----------------------------------+---------------+-------------+-----------+
-| 4f33a63031424ed095b8c2f9e8b15ecb |            16 |          16 |  10000005 |
-+----------------------------------+---------------+-------------+-----------+
+-- Compact the table table
+optimize table t compact;
+
+-- Notice that a new snapshot was added
+select snapshot_id, segment_count, block_count, timestamp from fuse_snapshot('default', 't');
+
+snapshot_id                     |segment_count|block_count|timestamp            |
+--------------------------------+-------------+-----------+---------------------+
+9828b23f74664ff3806f44bbc1925ea5|            1|          1|2022-12-26 19:39:27.0|
+edc9477b62c24f299c05a4d189030772|            3|          3|2022-12-26 19:33:49.0|
+256f1fe2e3974ee595474b2a8cad7a39|            2|          2|2022-12-26 19:33:42.0|
+1e060f7d145747578963b5a7e3b14742|            1|          1|2022-12-26 19:32:57.0|
+
+-- Purge the historical data generated before the new snapshot was created.
+optimize table t purge before (SNAPSHOT => '9828b23f74664ff3806f44bbc1925ea5');
+
+select snapshot_id, segment_count, block_count, timestamp from fuse_snapshot('default', 't');
+
+snapshot_id                     |segment_count|block_count|timestamp            |
```

Comment:
@dantengsky This result is supposed to be as follows according to https://github.com/datafuselabs/databend/pull/9318. But when I ran the example in my environment set up with [v0.8.161-nightly](https://github.com/datafuselabs/databend/releases/tag/v0.8.161-nightly), it still showed the same 4 snapshots as the ones before the purge. It seems the purge doesnt work. Can you help check why?

By the way, something else was NOT the same as you mentioned in https://github.com/datafuselabs/databend/pull/9318:

- if you run 'optimize table t purge', Databend doesnt report any syntax errror, and nothing seems to be done behind. 
- The same situation when i ran 'optimize table t all'.

Reply:
looks like a bug,  but I just tried v0.8.161-nighly musl build, and failed to reproduce the issue :

~~~
mysql> create table t(c int);
Query OK, 0 rows affected (0.55 sec)

mysql> select version();
+-------------------------------------------------------------------------------------+
| version()                                                                           |
+-------------------------------------------------------------------------------------+
| 8.0.26-v0.8.161-nightly-6f9a509(rust-1.67.0-nightly-2022-12-22T16:18:57.778904349Z) |
+-------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> insert into t values(1);
Query OK, 1 row affected (0.02 sec)

mysql> insert into t values(2);
Query OK, 1 row affected (0.02 sec)

mysql> insert into t values(3);
Query OK, 1 row affected (0.03 sec)

mysql> select snapshot_id, segment_count, block_count, timestamp from fuse_snapshot('default', 't');
+----------------------------------+---------------+-------------+----------------------------+
| snapshot_id                      | segment_count | block_count | timestamp                  |
+----------------------------------+---------------+-------------+----------------------------+
| a2cdda8f82074901a11ed29a75167473 |             3 |           3 | 2022-12-28 09:32:05.751158 |
| f0336af326af45d085463c9d4de8d4f0 |             2 |           2 | 2022-12-28 09:32:03.709165 |
| dc4eec26f1144f56ba5ece7f65b52e16 |             1 |           1 | 2022-12-28 09:32:01.759387 |
+----------------------------------+---------------+-------------+----------------------------+
3 rows in set (0.01 sec)
Read 3 rows, 594.00 B in 0.004 sec., 705.61 rows/sec., 136.44 KiB/sec.

mysql> optimize table t purge before (SNAPSHOT => 'a2cdda8f82074901a11ed29a75167473');
Query OK, 0 rows affected (0.01 sec)

mysql> select snapshot_id, segment_count, block_count, timestamp from fuse_snapshot('default', 't');
+----------------------------------+---------------+-------------+----------------------------+
| snapshot_id                      | segment_count | block_count | timestamp                  |
+----------------------------------+---------------+-------------+----------------------------+
| a2cdda8f82074901a11ed29a75167473 |             3 |           3 | 2022-12-28 09:32:05.751158 |
+----------------------------------+---------------+-------------+----------------------------+
1 row in set (0.01 sec)
Read 1 rows, 226.00 B in 0.004 sec., 254.4 rows/sec., 56.15 KiB/sec.

mysql> insert into t values(4);
Query OK, 1 row affected (0.02 sec)

mysql> optimize table t purge;
Query OK, 0 rows affected (0.02 sec)

mysql> select snapshot_id, segment_count, block_count, timestamp from fuse_snapshot('default', 't');
+----------------------------------+---------------+-------------+----------------------------+
| snapshot_id                      | segment_count | block_count | timestamp                  |
+----------------------------------+---------------+-------------+----------------------------+
| ac2256c99c664ec981bdb16157349065 |             4 |           4 | 2022-12-28 09:35:00.727919 |
+----------------------------------+---------------+-------------+----------------------------+
1 row in set (0.01 sec)
Read 1 rows, 226.00 B in 0.005 sec., 214.31 rows/sec., 47.30 KiB/sec.

mysql> insert into t values(5);
Query OK, 1 row affected (0.02 sec)

mysql> optimize table t all;
Query OK, 5 rows affected (0.04 sec)

mysql> select snapshot_id, segment_count, block_count, timestamp from fuse_snapshot('default', 't');
+----------------------------------+---------------+-------------+----------------------------+
| snapshot_id                      | segment_count | block_count | timestamp                  |
+----------------------------------+---------------+-------------+----------------------------+
| ff489aa9956a4842a3bfef801ea2c245 |             1 |           1 | 2022-12-28 09:35:39.049651 |
+----------------------------------+---------------+-------------+----------------------------+
1 row in set (0.01 sec)
Read 1 rows, 226.00 B in 0.004 sec., 244.71 rows/sec., 54.01 KiB/sec.

mysql> select count(*) from t;
+----------+
| count(*) |
+----------+
|        5 |
+----------+
1 row in set (0.01 sec)
Read 1 rows, 1.00 B in 0.001 sec., 1.11 thousand rows/sec., 1.08 KiB/sec.
~~~

Reply:
> By the way, something else was NOT the same as you mentioned in https://github.com/datafuselabs/databend/pull/9318:
> 
> if you run 'optimize table t purge', Databend doesnt report any syntax errror, and nothing seems to be done behind.

sorry, my description of that PR is vague, statement `optimize table purge`  (without any travel point specified) is still allowed, and it will purge all the historical data (except the latest snapshot).

i`optimize table t`  is not allowed:

~~~
mysql> optimize table t;
ERROR 1105 (HY000): Code: 1005, displayText = error:
  --> SQL:1:17
  |
1 | optimize table t
  | --------        ^ expected `ALL`, `PURGE`, `COMPACT`, or `.`
  | |
  | while parsing `OPTIMIZE TABLE [<database>.]<table> (ALL | PURGE | COMPACT [SEGMENT])`
~~~

Reply:
Ok, i just put these back to SYNTAX:
- OPTIMIZE TABLE [database.]table_name PURGE
- OPTIMIZE TABLE [database.]table_name ALL